### PR TITLE
Frontend update post comment

### PR DESCRIPTION
**Pull request by eeels22**
# Changes proposed in this pull request

- Toggle update button
- Update post (anyone)
- Update comment (anyone)
- Restructure component folders

## User stories
```
As a user of Cuddly,
I want to be able to edit my posts,
so that I can change something or add to my post. 
```
```
As a user of Cuddly,
I want to be able to update my comment,
so that I can change something or add to my comment. 
```

### CODE REVIEW

File: frontend/src/components/comments/CommentCard.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      <button onClick={handleDelete}>Delete</button>
+      <button onClick={() => (toggle ? setToggle(false) : setToggle(true))}>
+        {toggle ? "Cancel Update" : "Update"}
+      </button>
+      {toggle && (
+        <CommentUpdateForm
+          onSubmit={(commentData) => updateComment(commentData)}
+          comment={comment}
+        />
+      )}
```
Comment (NeuralAlchemist) [inline frontend/src/components/comments/CommentCard.jsx:37]: is this the logic that will show the update form or not? very simple and clever

File: frontend/src/components/comments/CommentUpdateForm.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    setContentText("");
+  };
+
+  return (
+    <form>
+      <div>
+        <div>
+          <textarea
+            value={contentText}
+            onChange={(e) => setContentText(e.target.value)}
```
Comment (NeuralAlchemist) [inline frontend/src/components/comments/CommentUpdateForm.jsx:23]: will this update the contentText for every time a change occurs? If so is it at all possible to update it only on submit?

File: frontend/src/components/posts/PostUpdateForm.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    setContentText("");
+  };
+
+  return (
+    <form>
+      <div>
+        <div>
+          <textarea
+            value={contentText}
+            onChange={(e) => setContentText(e.target.value)}
```
Comment (NeuralAlchemist) [inline frontend/src/components/posts/PostUpdateForm.jsx:23]: will this update the contentText for every time a change occurs? If so is it at all possible to update it only on submit?

File: frontend/src/components/comments/CommentCard.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      console.error(e);
+    }
+  }
+
+  return (
+    <div>
+      <p>{comment.contentText}</p>
+      <button onClick={handleDelete}>Delete</button>
+      <button onClick={() => (toggle ? setToggle(false) : setToggle(true))}>
+        {toggle ? "Cancel Update" : "Update"}
```
Comment (deciocandido) [inline frontend/src/components/comments/CommentCard.jsx:30]: We could just have Cancel instead of Cancel Update

**Review by NeuralAlchemist — APPROVED** [review]
great work on the refactor!

**Review by deciocandido — APPROVED** [review]

File: frontend/src/components/comments/CommentCard.jsx (hunk 1 above)
Comment (eeels22) [reply frontend/src/components/comments/CommentCard.jsx:37]: yes, inspired by code from BeeConnected

File: frontend/src/components/comments/CommentUpdateForm.jsx (hunk 2 above)
Comment (eeels22) [reply frontend/src/components/comments/CommentUpdateForm.jsx:23]: added as an enhancement issue

File: frontend/src/components/posts/PostUpdateForm.jsx (hunk 3 above)
Comment (eeels22) [reply frontend/src/components/posts/PostUpdateForm.jsx:23]: added as enhancement issue

File: frontend/src/components/comments/CommentCard.jsx (hunk 4 above)
Comment (eeels22) [reply frontend/src/components/comments/CommentCard.jsx:30]: We thought we would use it just as placeholder text as it's likely the text will be replaced by icons

**Pull request merged**